frontend/src/scenes/cohorts/Cohorts.tsx

```diff
@@ -152,16 +152,19 @@ export function Cohorts(): JSX.Element {
                 <LemonInput
                     type="search"
                     placeholder="Search for cohorts"
-                    onChange={setSearchTerm}
+                    onChange={(search) => {
+                        setSearchTerm(search)
+                        setCohortFilters({ search: search || undefined, page: 1 })
+                    }}
                     value={searchTerm}
                 />
             </div>
             <LemonTable
                 columns={columns}
                 loading={cohortsLoading}
                 rowKey="id"
-                pagination={{ pageSize: 100 }}
-                dataSource={searchTerm ? cohortsSearch(searchTerm) : cohorts ?? []}
+                pagination={pagination}
+                dataSource={cohorts.results}
```

Comment:
the pagination and data are now controlled exclusively by the logic, and not some mix of react state management and the logic.  And now pagination actually works (it didn't beforehand 😓 )